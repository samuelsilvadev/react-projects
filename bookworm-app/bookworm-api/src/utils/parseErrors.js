module.exports = function parseErrors(errors) {
	const result = {};
	if (errors) {
		Object
			.entries(errors)
			.forEach((value) => {
				result[value[0]] = value[1].message;
			});
	}
	return result;
};
