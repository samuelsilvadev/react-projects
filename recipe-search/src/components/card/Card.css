.card {
	display: flex;
	flex-direction: column;
	box-shadow: 0 1px 3px 0 #d4d4d5;
	transition: transform 200ms;
}

.card:hover {
	transform: translateY(-5px);
}

.card__image {
	max-width: 100%;
	height: 300px;
	object-fit: cover;
}

.card__title {
	font-size: 18px;
	font-weight: 700;
	text-transform: uppercase;
	color: #d63031;
}

.card__subtitle {
	text-transform: uppercase;
	font-size: 14px;
}

.card__image,
.card__title,
.card__subtitle {
	margin-bottom: 10px;
}

.card__title,
.card__description,
.card__subtitle,
.card__action {
	padding: 0 10px;
}

.card__action {
	align-self: flex-start;
	border: none;
	box-shadow: 0 1px 3px 0 #d4d4d5;
	cursor: pointer;
	height: 30px;
	margin: 0 10px 10px;
}

.card__action--link {
	align-items: center;
	display: flex;
	color: currentColor;
	text-decoration: none;
}
