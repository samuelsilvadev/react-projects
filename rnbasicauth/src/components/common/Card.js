/**
 * @format
 * @flow
 */

import * as React from 'react';
import { StyleSheet, View } from 'react-native';

type Props = {
    children: any,
}

export function Card(props: Props) {
    const { children } = props;

    return (
        <View style={ styles.container }>
            { children }
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        borderWidth: 1,
        borderRadius: 2,
        borderColor: '#ddd',
        borderBottomWidth: 0,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.1,
        shadowRadius: 2,
        elevation: 1,
        marginLeft: 5,
        marginRight: 5,
        marginTop: 10,
    },
});