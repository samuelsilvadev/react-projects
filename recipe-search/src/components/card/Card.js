import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './Card.css';

export const Card = ({ imageSrc, title, description, subtitle, action, link, ...rest }) => (
	<div className="card" {...rest}>
		{imageSrc && <img className="card__image" src={imageSrc} alt={title} data-test="card__image" />}
		{title && <h2 className="card__title" data-test="card__title">{title}</h2>}
		{subtitle && <p className="card__subtitle" data-test="card__subtitle">{subtitle}</p>}
		{description && <p className="card__description" data-test="card__description">{description}</p>}
		{action && <button type="button" className="card__action" onClick={action.onClick} data-test="card__button">{action.text}</button>}
		{link && <Link to={link.to} className="card__action card__action--link" data-test="card__link">{link.text}</Link>}
	</div>
);

Card.propTypes = {
	imageSrc: PropTypes.string,
	title: PropTypes.string,
	subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
	description: PropTypes.string,
	action: PropTypes.shape({
		text: PropTypes.string.isRequired,
		onClick: PropTypes.func.isRequired
	}),
	link: PropTypes.shape({
		text: PropTypes.string.isRequired,
		to: PropTypes.any.isRequired
	})
};

export default Card;
