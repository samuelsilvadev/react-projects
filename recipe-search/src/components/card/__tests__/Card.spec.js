import React from 'react';
import { shallow } from 'enzyme';

import Card from '../Card';

import { toTestId } from '@tests/utils';

describe('<Card />', () => {
	it('should render correctly with all props', () => {
		const wrapper = shallow(<Card
			imageSrc="./something.png"
			title="Hello"
			subtitle="world"
			description="Loren ipsum is awesome"
			action={{
				text: 'button',
				onClick: () => { }
			}}
			link={{
				to: 'somewhere.com',
				text: 'link'
			}} />);

		expect(wrapper.find(toTestId('card__title')).text()).toBe('Hello');
		expect(wrapper.find(toTestId('card__subtitle')).text()).toBe('world');
		expect(wrapper.find(toTestId('card__description')).text()).toBe('Loren ipsum is awesome');
		expect(wrapper.find(toTestId('card__button')).text()).toBe('button');
		expect(wrapper.find(toTestId('card__link')).childAt(0).text()).toBe('link');
		expect(wrapper).toMatchSnapshot();
	});

	it('should call `onClick` when click on button', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Card
			action={{
				text: 'button',
				onClick,
			}}
		/>);
		const button = wrapper.find(toTestId('card__button'));

		button.simulate('click');

		expect(onClick).toHaveBeenCalledTimes(1);
	});

	it('should render with image', () => {
		const wrapper = shallow(<Card
			imageSrc="./something.png"
			title="Hello"
		/>);
		const image = wrapper.find(toTestId('card__image'));

		expect(image.prop('src')).toBe('./something.png');
		expect(image.prop('alt')).toBe('Hello');
	});
});
