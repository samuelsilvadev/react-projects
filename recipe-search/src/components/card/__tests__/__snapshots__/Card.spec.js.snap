// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly with all props 1`] = `ShallowWrapper {}`;
